packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart

```diff
@@ -0,0 +1,248 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'ui.dart';
+
+/// Configuration options for the GoogleMaps user interface.
+@immutable
```

Comment:
Wouldn't every non-scalar member of the MapConfiguration need to be `@immutable` too, or their internal state could be mutated across diffs of a MapConfiguration?

It seems that `immutable` [doesn't enforce this](https://api.flutter.dev/flutter/meta/immutable-constant.html) (only that all are `final`)?

Reply:
That's an interesting question. I've personally used `@immutable` for two reasons:
- Prevent accidentally adding non-final fields later (which I've seen happen in some of our plugins), which is why I did it here.
- Allow implementing ==/hash without the linter complaining.

For the first it doesn't matter, but for the second (which doesn't apply here) it does. If you used an "immutable" object as a map key based on "hash", and a sub-field were mutated, it would end badly. I guess that's more a limitation of the lint than anything else?

I'll ask about this in the Dart Discord channel though because now I'm curious what the thinking on this is generally.